frontend/eda/rulebook-activations/components/SourceEventMapFields.tsx

```diff
@@ -0,0 +1,108 @@
+import { Button, FormFieldGroupExpandable, FormFieldGroupHeader } from '@patternfly/react-core';
+import { TrashIcon } from '@patternfly/react-icons';
+import { useTranslation } from 'react-i18next';
+import { useFormContext, useWatch } from 'react-hook-form';
+import { PageFormTextArea } from '../../../../framework';
+import { EdaSource, EdaSourceEventMapping } from '../../interfaces/EdaSource';
+import { edaAPI } from '../../common/eda-utils';
+import { EdaResult } from '../../interfaces/EdaResult';
+import { EdaRulebook } from '../../interfaces/EdaRulebook';
+import { useGet } from '../../../common/crud/useGet';
+import { EdaWebhook } from '../../interfaces/EdaWebhook';
+import { PageFormSingleSelect } from '../../../../framework/PageForm/Inputs/PageFormSingleSelect';
+
+export function SourceEventMapFields(props: {
+  index: number;
+  rulebook: EdaRulebook;
+  source_mappings: EdaSourceEventMapping;
+  onDelete: (id: number) => void;
+}) {
+  const { t } = useTranslation();
+  const { index, onDelete } = props;
+  const { register, setValue } = useFormContext();
+
+  const { data: sources } = useGet<EdaResult<EdaSource>>(
+    edaAPI`/rulebooks/` + `${props?.rulebook?.id}/sources/?page=1&page_size=200`
+  );
+  const { data: events } = useGet<EdaResult<EdaWebhook>>(edaAPI`/webhooks/?page=1&page_size=200`);
+
+  const selectedSource = useWatch({ name: `mappings.${index}.source_name` }) as string;
+  let srcIndex = -1;
+  if (sources?.results) {
+    srcIndex = sources.results.findIndex((source) => source.name === selectedSource);
+
+    if (srcIndex > -1) {
+      setValue(`${index}.source_info`, sources.results[srcIndex].source_info);
+      setValue(`mappings.${index}.rulebook_hash`, sources.results[srcIndex].rulebook_hash);
+    }
+  }
+  const selectedEvent = useWatch({ name: `mappings.${index}.webhook_id` }) as number;
+  let evIndex = -1;
+  if (events?.results) {
+    evIndex = events.results.findIndex((event) => event.id === selectedEvent);
+
+    if (evIndex > -1) {
+      setValue(`mappings.${index}.webhook_name`, events.results[evIndex].name);
+    }
+  }
+
+  return (
+    <FormFieldGroupExpandable
+      isExpanded
+      header={
+        <FormFieldGroupHeader
+          titleText={{ text: t('Mapping ') + `${index + 1}`, id: `Mapping ${index}` }}
```

Comment:
We could translate this without concatenating with interpolation: ` t('Mapping {{count}}', { count: index + 1 })`